### ISSUE TITLE
socks auth dont work in smart mode

### ISSUE THREAD

**Issue by odo125** [salarcode/SmartProxy#47]
If I use a socks5 proxy with authentification with the setting "Always Enable" it works, but with the same proxysettings it does not work with the setting "Smart Proxy"
 I checked this with my own Ubuntu server running a dante socks proxy with authentification.
The error logs of the socks server say:
" error after reading 3 bytes in 0 seconds: client offered no acceptable authentication method"

**Comment by salarcode** [comment]
Did you add a rule for the site you want to have proxy applied?
SmartMode works with rules

**Comment by odo125** [comment]
Yes Rule for setting Smart Proxy added and it works if I dont need authentification, but if I turn on auth on the dante socks server it dont work anymore. If I use the setting Always enable it works with and without Authentification.

**Comment by salarcode** [comment]
Can you please test a special version of smartproxy to see if recent patches work for you? that would be great help.

1- First, disable SmartProxy existing in you browser
2 - Go to releases page and download and install the "unlisted" version of the extension.
	https://github.com/salarcode/SmartProxy/releases/download/v0.7.3/smartproxy_unlisted-firefox-0.7.3-an.fx-windows.xpi
3- Apply your servers/rules and configuration by entering them or restoring settings backup.
4- Test your issues if they are fixed

Thanks

**Comment by odo125** [comment]
Sorry to inform you that also with your downloaded version my issues are still there. The Log output of socks 5 server is: 
Jan 16 10:55:12 (1516096512.666136) danted[18439]: info: block(1): tcp/accept ]: 192.168.0.2.58883 192.168.0.1.1080: error after reading 3 bytes in 0 seconds: client offered no acceptable authentication method.
If I go to the site with "Immer aktiv" which is the same as in previous version "always enable" the log output of socks 5 server is:
Jan 16 11:00:46 (1516096846.797869) danted[18439]: info: pass(1): tcp/accept [: 192.168.0.2.58930 192.168.0.1.1080
Jan 16 11:00:46 (1516096846.892826) danted[18128]: info: pass(1): tcp/connect [: username%root@192.168.0.2.58930 192.168.0.1.1080 -> 10.7.0.2.58930 www.wieistmeineip.at.443
Jan 16 11:00:47 (1516096847.093474) danted[18441]: info: pass(1): tcp/accept [: 192.168.0.2.58931 192.168.0.1.1080
Thanks for reviewing hope the information helps

**Comment by salarcode** [comment]
Thank you I will look into it

**Comment by salarcode** [comment]
It can be an issue in you proxy 'Dante' proxy implementation. I've tested with 3 different proxies and just one of them failed which was my own written one.
You can try CCProxy on windows to see if that makes any difference

**Comment by odo125** [comment]
Ok I think I have found the problem. I had to delete all smart proxy rules, then change the socks proxy and after this applying all the rules again. Then everything works as expected. I realized this when I used the CCProxy on a Windows machine. There I saw in the logs that smart proxy with smart mode enabled did not use the ccproxy but it still uses the dante proxy on the linux machine. So the rules were not updated if you change the socks proxy
